addons/autoLock/autoLock.lua

```diff
@@ -0,0 +1,166 @@
+--   used to endorse or promote products derived from this software without
+--   specific prior written permission.
+
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+-- ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+-- WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+-- DISCLAIMED. IN NO EVENT SHALL ReaperX BE LIABLE FOR ANY
+-- DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+-- (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+-- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+-- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+-- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+--
+--
+-- Automatically equips a gear set and uses the /lockstyle command when leaving town.
+-- This addon requires the Gearswap addon.
+--
+-- To use this addon create a gear set titled sets.lockstyle in your job.lua file.
+-- Ex:
+-- 		sets.lockstyle = {
+--			head="Raider's Bonnet +2",
+--			body="Raider's Vest +2",hands="Raider's Armlets +2",
+--			legs="Raider's Culottes +2",feet="Raider's Poulaines +2"}
+--
+--
+
+_addon.name = "autoLock"
+_addon.author = "bangerang"
+_addon.version = "1.0"
+_addon.commands = {'al', 'autolock'}
+_addon.language = 'english'
+
+require('luau')
+config = require('config')
+
+defaults = {}
+defaults.zoneDelay = 10
+defaults.keyBind = '^x'
+defaults.autoLock = true
+defaults.ignoreZones = S{
+    "Ru'Lude Gardens", "Upper Jeuno", "Lower Jeuno", "Port Jeuno", "Port Windurst", "Windurst Waters", "Windurst Woods", "Windurst Walls",
+    "Heavens Tower", "Port San d'Oria", "Northern San d'Oria", "Southern San d'Oria", "Chateau d'Oraguille", "Port Bastok", "Bastok Markets",
+    "Bastok Mines", "Metalworks", "Aht Urhgan Whitegate", "Tavnazian Safehold", "Nashmau", "Selbina", "Mhaura", "Norg", "Rabao", "Kazham",
+    "Eastern Adoulin", "Western Adoulin", "Leafallia", "Celennia Memorial Library", "Mog Garden"}
+
+settings = config.load(defaults)
+
+tokens = {}
+tokens.lockstyle = false
+
+windower.send_command('bind %s al':format(settings.keyBind))
+windower.send_command('input /lockstyle off')
+
+function al_output(msg)
+    prefix = 'autoLock: '
+    windower.add_to_chat(209, prefix..msg)
+end
+
+-- Accepts a boolean value and returns an appropriate string value. i.e. true -> 'on'
+function booltostr(bool)
+    local str = ''
+    if bool then
+        str = 'on'
+        return str
+    elseif not bool then
+        str = 'off'
+        return str
+    else
+        print('Boolean to string: unknown error.')
+    end
+end
+
+function command_lockstyle()
+	if tokens.lockstyle then
+		windower.send_command('input /lockstyle off')
+		coroutine.sleep(1)
+	end
+	windower.send_command('gs equip sets.lockstyle;wait 1.5;input /lockstyle on')
```

Comment:
You could make this configurable by the user instead of forcing them to make a set called `sets.lockstyle`. Just add `defaults.LockSet = 'sets.lockstyle'` to the default settings then users can overwrite it if they want.